.gitignore

```diff
@@ -1,3 +1,10 @@
+
+*.idea
+*.Python
+*.iml
+*target
+*.pyc
+*.DS_Store
```

Comment:
Don't checkin .pyc files. You included it in .gitignore and still .pyc file is seen in checkin.
See below for review comments. 
https://docs.google.com/spreadsheets/d/1hqu6WtRwc5FxdsyWuhOV1RY0LfSTTLluMXByBHWqCBY/edit#gid=56856103